libs/sdk-core/src/error.rs

```diff
@@ -201,6 +201,52 @@ impl From<PersistError> for ReceiveOnchainError {
     }
 }
 
+pub type RedeemOnchainResult<T, E = RedeemOnchainError> = Result<T, E>;
+
+#[derive(Debug, Error)]
+pub enum RedeemOnchainError {
+    /// This error is raised when a general error occurs not specific to other error variants
+    /// in this enum.
+    #[error("Generic: {err}")]
+    Generic { err: String },
+
+    /// This error is raised when a connection to an external service fails.
+    #[error("Service connectivity: {err}")]
+    ServiceConnectivity { err: String },
+
+    // This error is raised when the node does not have enough funds to redeem the onchain balance.
+    #[error("{err}")]
+    InsufficientFunds { err: String },
+}
+
+impl From<NodeError> for RedeemOnchainError {
+    fn from(value: NodeError) -> Self {
+        match value {
+            NodeError::InsufficientFunds(err) => Self::InsufficientFunds { err },
```

Comment:
Could convert ServiceConnectivity here also
```suggestion
            NodeError::InsufficientFunds(err) => Self::InsufficientFunds { err },
            NodeError::ServiceConnectivity(err) => Self::ServiceConnectivity { err },
```